### PR TITLE
CORTX-30097: Fix codacy issues.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

# Problem Statement
- Fix codacy issue for cortx-rgw-integration repo
Codacy result for this PR: https://app.codacy.com/gh/Seagate/cortx-rgw-integration/dashboard?branch=fix-codacy-issue

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/rgw/setup/rgw.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 import time
 import errno
 import json
-import socket
+#import socket
```
Comment (sachitanands) [inline src/rgw/setup/rgw.py:21]: pls remove this import

Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py]: done

**Review by rahul-tripathi-git — APPROVED** [review]

File: src/rgw/setup/rgw.py [hunk 2]
```diff
@@ -609,6 +577,44 @@ def _update_hax_endpoint_and_create_admin(conf: MappedConf):
                         raise SetupError(status, 'Admin user creation failed on'
                             f' "{Rgw._machine_id}" node, with all data pods - {data_pod_hostnames}')
 
+    @staticmethod
+    def _get_lock(consul_idx: str):
+        """Aquire consul lock."""
```
Comment (rahul-tripathi-git) [inline src/rgw/setup/rgw.py:582]: """Get lock from consul kv"""

Comment (rahul-tripathi-git) [reply src/rgw/setup/rgw.py:582]: """Aquire consul lock."""
Sounds like we are using ACLs from consul. Which is not the case here.

Comment (shriya-deshmukh) [reply src/rgw/setup/rgw.py]: Done.

**Comment by shriya-deshmukh** [comment]
Codacy result - https://app.codacy.com/gh/Seagate/cortx-rgw-integration/dashboard?branch=fix-codacy-issue

File: src/rgw/setup/rgw_service.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def start(conf: MappedConf, config_file, log_file, motr_trace_dir, index: str =
             args = shlex.split(args)
             sys.stdout.flush()
             sys.stderr.flush()
-            os.execl(cmd, cmd, *args)
+            os.execl(cmd, cmd, *args)  # nosec
```
Comment (cdeshmukh) [inline src/rgw/setup/rgw_service.py:38]: this is suppressing the warning. what justification is there to do same ? @shriya-deshmukh

Comment (shriya-deshmukh) [reply src/rgw/setup/rgw_service.py:38]: We need  radosgw service to run as main process in container, So this execl is replacing rgw_setup  calling process with our radosgw process. which is not possible through other ways (os.system, subprocess.run subpeocess.Popen etc)
So we have kept os.execl and disable the codacy warning for this line.

Comment (cdeshmukh) [reply src/rgw/setup/rgw_service.py:38]: Ok i remember now. Also, wanted to capture reason here

**Pull request merged**